server/src/main/java/org/opensearch/action/support/RetryableAction.java

```diff
@@ -74,7 +76,15 @@ public RetryableAction(
         TimeValue timeoutValue,
         ActionListener<Response> listener
     ) {
-        this(logger, threadPool, initialDelay, timeoutValue, listener, ThreadPool.Names.SAME);
+        this(
+            logger,
+            threadPool,
+            initialDelay,
+            timeoutValue,
+            listener,
+            BackoffPolicy.exponentialRandomBackoff(initialDelay.getMillis()),
+            ThreadPool.Names.SAME
```

Comment:
which is this threadpool?

Reply:
As per my understanding, This is basically caller's threadpool only. We will not create new threadpool for retries but it will perform the retries on the same threadpool of the caller's.